linera-core/src/chain_worker/state.rs

```diff
@@ -1055,10 +1078,25 @@ where
             tip.block_hash == block.previous_block_hash,
             WorkerError::InvalidBlockChaining
         );
+
+        let mut blob_ids_in_oracle_records = HashSet::new();
+        for record in oracle_records {
+            for response in &record.responses {
+                if let OracleResponse::Blob(blob_id) = response {
+                    blob_ids_in_oracle_records.insert(*blob_id);
+                }
+            }
+        }
```

Comment:
Maybe we should move this into `impl ExecutedBlock`, since `process_confirmed_block` is already pretty long:

```suggestion
        let required_blob_ids = executed_block.required_blob_ids();
```